# Bioinformatics_Melanoma
<img width="1281" alt="Screenshot 2024-02-12 at 11 54 40â€¯AM" src="https://github.com/ZelihaB/Bioinformatics_Melanoma/assets/71549587/feb67064-f271-494c-bba4-673b62f45cc4">

## 1.Quality Control with FastQC

![diagram](https://github.com/ZelihaB/Bioinformatics_Melanoma/assets/71549587/591da8c0-809a-49dc-a04d-1e75a3b19dfe)

## 2. Trimming and Filtering with Trimmomatic and Cutadapt

![diagram(2)](https://github.com/ZelihaB/Bioinformatics_Melanoma/assets/71549587/465186cc-d6f3-48a8-b22f-aa23e9f70cb8)
![diagram(3)](https://github.com/ZelihaB/Bioinformatics_Melanoma/assets/71549587/2c8fce1c-e1df-4eff-a452-9f6b6efa0a93)

## 3.Reference Genome(GRCh38) Indexing with BWA

![diagram(4)](https://github.com/ZelihaB/Bioinformatics_Melanoma/assets/71549587/3b0bb650-7b6e-4b12-be7b-2c67f52046be)

## 4.Read Mapping with BWA mem
![diagram(5)](https://github.com/ZelihaB/Bioinformatics_Melanoma/assets/71549587/202a8b7f-1ca2-44fc-91af-51d54941571f)

## 5.Variant Calling with BCFtools or BATK
![diagram(6)](https://github.com/ZelihaB/Bioinformatics_Melanoma/assets/71549587/f3896af3-ce83-4721-a384-d6a3cb090279)

Detailed Steps:
   1. Install Homebrew:
    * Use the command /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)" to install Homebrew on macOS.
2. Install BCFtools and GATK:
    * For BCFtools: brew install bcftools
    * For GATK (and Java if needed): brew install --cask gatk and brew install openjdk
3. Prepare Data (BAM files):
    * Ensure your sequencing data is in BAM format, sorted and indexed.
4. Pre-process Data (for GATK):
    * Use tools like Picard and GATK's BaseRecalibrator for marking duplicates, realigning indels, and recalibrating base quality scores.
5. Call Variants:
    * With BCFtools: Use bcftools mpileup followed by bcftools call to call variants.
    * With GATK: Use gatk HaplotypeCaller specifying the reference genome -R, input BAM -I, and output VCF -O.
6. Filter Variants:
    * For BCFtools: Apply filters directly in the bcftools call command or use bcftools filter.
    * For GATK: Use gatk VariantFiltration to apply specific filters based on quality metrics.
7. Analysis Complete:
    * Review the VCF files generated by BCFtools or GATK for your variant analysis.

## 6. Annotate Variants with SpEff
![diagram(7)](https://github.com/ZelihaB/Bioinformatics_Melanoma/assets/71549587/1de5f285-b605-43ec-9641-c77c2f77c050)

## 7. Visualization and Interpretation with IGV
![diagram(8)](https://github.com/ZelihaB/Bioinformatics_Melanoma/assets/71549587/61ffc77e-e29a-4385-a13a-b1b255639c79)


